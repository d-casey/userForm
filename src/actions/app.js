import {
  RESET_FORM_STAGE,
  UPDATE_STAGE
} from '../constants/actions'

export const updateStage = stage => ({
  type: UPDATE_STAGE,
  payload: {
    stage
  }
})

export const resetFormStage = () => ({
  type: RESET_FORM_STAGE
})
