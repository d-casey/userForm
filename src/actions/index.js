import {
  SUBMIT_NEW_USER_FORM,
  SUBMIT_PRIVACY_FORM
} from '../constants/actions'

export const submitNewUserForm = formFields => ({ type: SUBMIT_NEW_USER_FORM, formFields })
export const submitPrivacyForm = formFields => ({ type: SUBMIT_PRIVACY_FORM, formFields })
