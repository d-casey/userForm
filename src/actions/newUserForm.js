import {
  RESET_USER_FORM,
  SUBMIT_NEW_USER_FORM
} from '../constants/actions'

export const submitNewUserForm = formFields => ({
  type: SUBMIT_NEW_USER_FORM,
  payload: formFields
})

export const resetUserForm = () => ({
  type: RESET_USER_FORM
})
