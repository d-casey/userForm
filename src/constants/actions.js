export const SUBMIT_NEW_USER_FORM = 'SUBMIT_NEW_USER_FORM'
export const SUBMIT_PRIVACY_FORM = 'SUBMIT_PRIVACY_FORM'
export const NAVIGATE = 'NAVIGATE'
