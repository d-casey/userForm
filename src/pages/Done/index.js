import React from 'react'
import { useSelector, useDispatch } from "react-redux"

const Done = () => {
  const myState = useSelector(state => state)
  console.log('DONE PAGE STATE', myState)
  return (
    <div>
      <p>Please verify your email address, you should have received an email from us already!</p>
    </div>
  )
}

export default Done
