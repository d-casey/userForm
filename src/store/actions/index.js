import {
  RESET_FORM_STAGE
} from '../constants/actions'

export const resetFormStage = () => ({ type: RESET_FORM_STAGE })

export * from './app'
export * from './newUserForm'
export * from './privacyForm'
