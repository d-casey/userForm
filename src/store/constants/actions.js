export const UPDATE_STAGE = 'UPDATE_STAGE'
export const UPDATE_USER_FIELDS = 'UPDATE_USER_FIELDS'
export const SUBMIT_PRIVACY_FORM = 'SUBMIT_PRIVACY_FORM'
export const RESET_USER_FORM = 'RESET_USER_FORM'
export const RESET_PRIVACY_FORM = 'RESET_PRIVACY_FORM'
export const RESET_FORM_STAGE = 'RESET_FORM_STAGE'
export const LOGGING_DETAILS = 'LOGGING_DETAILS'
export const RESET_ALL_FORMS = 'RESET_ALL_FORMS'
export const SET_FORM_COMPLETION = 'SET_FORM_COMPLETION'
export const RESET_APP = 'RESET_APP'
