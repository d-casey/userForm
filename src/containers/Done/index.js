import React, { useEffect } from 'react'
import { useSelector, useDispatch } from "react-redux"
import { logSubmittedFormDetails } from '../../store/actions'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

const Done = () => {
  const dispatch = useDispatch()
  const isFormCompleted = useSelector(state => state.app.formCompleted)
  const newUserFormData = useSelector(state => state.newUserForm)
  const privacyFormData = useSelector(state => state.privacyForm)
  const finalFormData = { newUserFormData, privacyFormData }

  useEffect(() => {
    if(!isFormCompleted) dispatch(logSubmittedFormDetails(finalFormData))
  })

  return (
    <div className="success">
      <FontAwesomeIcon className="fa-5x" icon="check" />
      <p>Please verify your email address, you should have received an email from us already!</p>
    </div>
  )
}

export default Done
