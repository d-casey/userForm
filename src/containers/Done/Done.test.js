import React from 'react'
import Done from '.'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { logSubmittedFormDetails } from '../../store/actions'

const mockStore = configureStore([])
let store
let component
let completionState = {
  app: {
    stage: 2,
    formCompleted: true
  },
  newUserForm: {
    name: "Severus Snape",
    role: "Potions Master",
    email: "thehalfbloodprince@owlmail.com",
    password: "ILoveP0tion5",
    isSubmitted: true
  },
  privacyForm : {
    isSubmitted: true,
    receive_updates: false,
    receive_communication: false,
  }
}

const getWrapper = () => {
  store = mockStore(completionState)
  component = renderer.create(
    <Provider store={store}>
      <Done />
    </Provider>
  )
}

describe('<Done /> container', () => {
  beforeEach(() => {
    getWrapper()
  })

  it('renders the success content as expected', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
