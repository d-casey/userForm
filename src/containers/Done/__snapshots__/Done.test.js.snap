// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Done /> container renders the success content as expected 1`] = `
<div
  className="form-success"
>
  <p>
    Please verify your email address, you should have received an email from us already!
  </p>
</div>
`;
